### ISSUE TITLE
Disallow dynamic properties

### ISSUE THREAD

**Issue by rentalhost** [kalessil/phpinspectionsea#253]
This should warn if user call for read or write on a not declared property. Maybe related to #250.

```php
public function __construct() {
    // Warning: $property was not declared on this class.
    $this->property = 123;
}
```

Should have two ways to avoid that:

1. Declare as a property: `public/protected/private $property`.
2. Declare as a `@property`: `@property $property`.

**Comment by kalessil** [comment]
Is this constructor specific only?

**Comment by rentalhost** [comment]
I guess that it can be valid in any method (or even outside class).

**Comment by kalessil** [comment]
- property annotations ignored;
- magic methods (__get/__set) are taken into account;

2 use cases to target:
- property annotations + magic methods (from parent/trait);
- implicit property definition;

**Comment by funivan** [comment]
This inspection already exists in phpstorm: `php -> undefined field` :)

**Comment by rentalhost** [comment]
Oh, really. I have disabled that for some reason that I can remember. I don't know if @kalessil annotations can mean something now.

**Comment by kalessil** [comment]
Will not happen - PS has already similar inspection
